server/action_base.h

```diff
@@ -156,6 +156,8 @@ class Action {
   bool action_uses_cleanup;
   bool cleanup_started;  // action has started cleanup process.
 
+  std::map<std::string, std::string> invalid_args;
```

Comment:
Why is a map necessary? Do you have an example where there are multiple arguments returned?

Reply:
I don't have such examples or may be there are none as we should return error as soon as we detect invalid argument. Now I feel it's not necessary yet it act as a container to hold the invalid argument. Should I go ahead and replace it with two different strings to hold arg name and arg value?

Reply:
Replaced std::map with two strings to hold arg name and arg value.

Reply:
You could also use an std::pair, but it's fine.